src/views/@common/components/ScrollToTop.tsx

```diff
@@ -0,0 +1,17 @@
+import { ReactNode, useEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+
+interface ScrollToTopProps {
+  children: ReactNode;
+}
+
+const ScrollToTop = (props: ScrollToTopProps) => {
+  const { pathname } = useLocation();
+
+  useEffect(() => {
+    window.scrollTo(0, 0);
+  }, [pathname]);
+  return <div>{props.children}</div>;
+};
```

Comment:
ScrollToTop 아주 잘 만들어주었는데요! 
해당 로직은 pathname이 바뀜에 따라 window의 스크롤 위치를 바꿔주는 코드이기 때문에 
꼭 적용시킬 내부 컴포넌트들을 children으로 들고있을 필요가 없습니다! 
따라서 아래와 같이 별다른 props나 return 없이 의도한 로직만을 담고, 
적용해주고 싶은 컴포넌트에서도 최상단에서 감싸줄 필요 없이 <ScrollToTop/> 이렇게 컴포넌트 내부에 추가만 해주어도 잘 작동한답니다! 

```suggestion
const ScrollToTop = () => {
  const { pathname } = useLocation();

  useEffect(() => {
    window.scrollTo(0, 0);
  }, [pathname]);
  return null;
};
```
적용시키는 컴포넌트인 CheckOfferPage에서도 code Suggestion 추가해놓았습니다!